tangelo/linq/simulator.py

```diff
@@ -520,21 +372,66 @@ def _statevector_to_frequencies(self, statevector):
                 this_chunk = self.n_shots % chunk_size if i == n_chunks else chunk_size
                 samples = distr.rvs(size=this_chunk)
                 freqs_shots += Counter(samples)
-            freqs_shots = {self.__int_to_binstr_lsq(k, n_qubits): v / self.n_shots for k, v in freqs_shots.items()}
+            freqs_shots = {self._int_to_binstr_lsq(k, n_qubits): v / self.n_shots for k, v in freqs_shots.items()}
             return freqs_shots
 
-    def __int_to_binstr(self, i, n_qubits):
+    def _int_to_binstr(self, i, n_qubits):
         """Convert an integer into a bit string of size n_qubits, in the order
         specified for the state vector.
         """
         bs = bin(i).split('b')[-1]
         state_binstr = "0" * (n_qubits - len(bs)) + bs
         return state_binstr[::-1] if (self.statevector_order == "msq_first") else state_binstr
 
-    def __int_to_binstr_lsq(self, i, n_qubits):
+    def _int_to_binstr_lsq(self, i, n_qubits):
         """Convert an integer into a bit string of size n_qubits, in the
         least-significant qubit order.
         """
         bs = bin(i).split('b')[-1]
         state_binstr = "0" * (n_qubits - len(bs)) + bs
         return state_binstr[::-1]
+
+    @property
+    def backend_info(self):
+        """A dictionary that includes {'noisy_simulation': True or False,
+                                       'statevector_available': True or False,
+                                       'statevector_order': 'lsq_first' or 'msq_first'"""
+        pass
+
+
+def Simulator(target=default_simulator, n_shots=None, noise_model=None, **kwargs) -> SimulatorBase:
```

Comment:
Will we keep this wrapper function in the future?

Reply:
We need a single entry-point to all our specialization of simulators, so we need a "Simulator" umbrella.